### DISCUSSION TITLE
Securing Cloudflare Account Access After tunnel creation (cert.pem file)

### DISCUSSION THREAD

**Discussion by rockinonthehubnotkansas** [homeassistant-apps/app-cloudflared#271] [Q&A] [answered]
I am looking for more information on the in which a compromised cert.pem file could lead to unauthorized Cloudflare account access. I noticed that this information is disclosed in the Troubleshooting section which would likely be read after the tunnel is already created (as is my case). 

Is it possible take the step of using another account to authorize access after the tunnel is already established using the main Cloudflare account? Or is it pointless at this juncture?
Is the current cert.pem file overwritten if setup is performed again with a different Cloudflare account?
Or, does removing the Cloudflared addon delete the cert.pem file? If so, is this enough to mitigate exposure of the cert.pem file?

In summary is there anything that can be done to secure Cloudflare access, or is it too late if you've already established the tunnel?

Thank you in advance.

**Comment by brenner-tobias** [comment, marked as answer]
Hi @rockinonthehubnotkansas,

this is an issue that I am planning to look into in more detail in a couple of weeks, since I am currently not home. But I can say that if you uninstall the add-on, the cert.pem file will be deletes, so this might be a way to move forward.

**Reply by rockinonthehubnotkansas** [reply]
Thank you for the affirmation that the cert.pem file is deleted upon uninstalling the addon. I am working on using the work around once Cloudflare fixes the ability add members.
